packages/repack/src/commands/common/runAdbReverse.ts

```diff
@@ -1,35 +1,98 @@
+import path from 'node:path';
 import execa from 'execa';
 import type { Logger } from '../../types';
 
 interface RunAdbReverseParams {
+  logger?: Logger;
   port: number;
   verbose?: boolean;
-  logger?: Logger;
+  wait?: boolean;
+}
+
+function getAdbPath() {
+  const androidHome = process.env.ANDROID_HOME;
+  return androidHome ? path.join(androidHome, 'platform-tools', 'adb') : 'adb';
+}
+
+function parseAdbError(error: unknown) {
+  const errorMessage = (error as Error).message;
+  const message = errorMessage.includes('error:')
+    ? errorMessage.split('error:')[1]
+    : errorMessage;
+  return message.trim();
+}
+
+async function executeAdbCommand(command: string, logger: Logger) {
+  const adbPath = getAdbPath();
+  try {
+    const result = await execa.command(`${adbPath} ${command}`);
+    logger.debug(`[ADB] "adb ${command}" executed successfully.`);
+    return result;
+  } catch (error) {
+    const message = parseAdbError(error);
+    logger.debug(`[ADB] "adb ${command}" failed: "${message}"`);
+    throw new Error(message);
+  }
+}
+
+async function waitForDevice(logger: Logger) {
+  try {
+    await executeAdbCommand('wait-for-device', logger);
+  } catch (error) {
+    const message = (error as Error).message;
+    if (/more than one device\/emulator/.test(message)) {
+      return;
```

Comment:
add a comment that this is expected and handled later

Reply:
done 👍